nameko/testing/pytest.py

```diff
@@ -84,30 +85,51 @@ def rabbit_manager(request):
 
 @pytest.yield_fixture()
 def rabbit_config(request, rabbit_manager):
+    import random
+    import string
     from kombu import pools
-    from nameko.testing.utils import (
-        reset_rabbit_vhost, reset_rabbit_connections,
-        get_rabbit_connections, get_rabbit_config)
+    from six.moves.urllib.parse import urlparse  # pylint: disable=E0611
+    from nameko.testing.utils import get_rabbit_connections
 
     amqp_uri = request.config.getoption('AMQP_URI')
 
-    conf = get_rabbit_config(amqp_uri)
+    uri = urlparse(amqp_uri)
+    username = uri.username
+    vhost = uri.path[1:]
 
-    reset_rabbit_connections(conf['vhost'], rabbit_manager)
-    reset_rabbit_vhost(conf['vhost'], conf['username'], rabbit_manager)
+    use_random_vost = (vhost == ":random:")
+
+    if use_random_vost:
+        vhost = "test_{}".format(
+            "".join(random.choice(string.ascii_lowercase) for _ in range(10))
+        )
+        amqp_uri = "{}://{}/{}".format(uri.scheme, uri.netloc, vhost)
+        rabbit_manager.create_vhost(vhost)
+        rabbit_manager.set_vhost_permissions(vhost, username, '.*', '.*', '.*')
+
+    conf = {
+        'AMQP_URI': amqp_uri,
+        'username': username,
+        'vhost': vhost
+    }
 
     yield conf
 
     pools.reset()  # close connections in pools
```

Comment:
it would feel better if this action raised something like a `ResetError` rather than raising a `RuntimeError` based on your count.